File: sb_payment_sdk/src/main/java/com/startbutton/sb_payment_sdk/models/APIResponse.java
package com.startbutton.sb_payment_sdk.models;

import com.google.gson.annotations.SerializedName;

public class APIResponse {
    @SerializedName("success")
    private boolean error;
    @SerializedName("message")
    private String message;
    @SerializedName("data")
    private Object data;

    public APIResponse(){
        this.error = true;
    }

    public APIResponse(boolean error, String message, Object data) {
        this.error = error;
        this.message = message;
        this.data = data;
    }

    public boolean getError() {
        return error;
    }

    public void setError(boolean error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }
}

File: lib/models/api_response.dart
class APIResponse<T>{
  var data;
  bool error;
  var message;

  APIResponse({this.data, this.error=false, this.message});

  factory APIResponse.fromJson(Map<String, dynamic> data){
    return APIResponse(
        error: data['status'] != null ? !data['status'] : data['error'] ?? false,
        message: data['message'],
        data: data['data']
    );
  }
}